tests/Umbraco.Tests.Integration/Umbraco.Infrastructure/PropertyEditors/BlockGridElementLevelVariationTests.cs

```diff
@@ -530,4 +530,239 @@ private IContent CreateContent(IContentType contentType, BlockGridValue blockGri
         ContentService.Save(content);
         return content;
     }
+
+    [Test]
+    public async Task Publishing_After_Changing_Element_Property_From_Variant_To_Invariant_Does_Not_Keep_Old_Culture_Specific_Values()
+    {
+        // 1. Create element type WITH culture variation
+        var elementType = CreateElementType(ContentVariation.Culture);
+        var areaKey = Guid.NewGuid();
+        var blockGridDataType = await CreateBlockGridDataType(elementType, areaKey);
+        var contentType = CreateContentType(blockGridDataType);
+
+        // 2. Create a simple block grid value with a single block for clarity
+        var contentElementKey = Guid.NewGuid();
+        var settingsElementKey = Guid.NewGuid();
+        var blockGridValue = new BlockGridValue([
+            new BlockGridLayoutItem(contentElementKey, settingsElementKey) { ColumnSpan = 12, RowSpan = 1 }
+        ])
+        {
+            ContentData =
+            [
+                new(contentElementKey, elementType.Key, elementType.Alias)
+                {
+                    Values =
+                    [
+                        new() { Alias = "invariantText", Value = "The invariant content value" },
+                        new() { Alias = "variantText", Value = "The content value in English", Culture = "en-US" },
+                        new() { Alias = "variantText", Value = "The content value in Danish", Culture = "da-DK" }
+                    ]
+                }
+            ],
+            SettingsData =
+            [
+                new(settingsElementKey, elementType.Key, elementType.Alias)
+                {
+                    Values =
+                    [
+                        new() { Alias = "invariantText", Value = "The invariant settings value" },
+                        new() { Alias = "variantText", Value = "The settings value in English", Culture = "en-US" },
+                        new() { Alias = "variantText", Value = "The settings value in Danish", Culture = "da-DK" }
+                    ]
+                }
+            ],
+            Expose =
+            [
+                new(contentElementKey, "en-US", null),
+                new(contentElementKey, "da-DK", null)
+            ]
+        };
+
+        var content = CreateContent(contentType, blockGridValue);
+        PublishContent(content, ["en-US", "da-DK"]);
+
+        // 3. Change element property type to invariant (remove culture variation)
+        foreach (var propertyType in elementType.PropertyTypes.Where(pt => pt.Alias == "variantText"))
+        {
+            propertyType.Variations = ContentVariation.Nothing;
+        }
```

Comment:
```suggestion
        elementType.PropertyTypes.Single(pt => pt.Alias == "variantText").Variations = ContentVariation.Nothing;
```

Seems we don't need the `foreach` here, as property aliases are unique.